en/docs/extended_topics/python_kernel_op.md

```diff
@@ -0,0 +1,291 @@
+- `op_type_name_py_kernel.py`(the `user_sigmoid_py_kernel.py` above) stores forward and backward calculation logic for custom operators implemented by Python.
+
+In the following sections:
+
+- How to write `op_type_name_cpp_def.cpp` to define Op.
+- How to write `op_type_name_py_api.py` to encapsulate the Python interface of Op.
+- How to write the `op_type_name_py_kernel.py` and uses Python to implement Op's calculation Kernel.
+- How to use `Python Kernel` custom Op in OneFlow.
+
+In the following section, we'll show how to implement a custom `user_relu` Op in Python.
+
+## Implementation and Registration of Op
+First, we define op in `user_relu_cpp_def.cpp` and complete the registration.
+```cpp
+#include "oneflow/core/framework/framework.h"
+
+namespace oneflow {
+namespace {
+
+REGISTER_USER_OP("user_relu_forward")
+  .Attr<std::string>("device_sub_tag", "py")
+  .Input("in")
+  .Output("out")
+  .SetTensorDescInferFn(
+      [](user_op::InferContext *ctx) -> Maybe<void> {
+        *ctx->Shape4ArgNameAndIndex("out", 0) =
+            *ctx->Shape4ArgNameAndIndex("in", 0);
+        *ctx->Dtype4ArgNameAndIndex("out", 0) =
+            *ctx->Dtype4ArgNameAndIndex("in", 0);
+        return Maybe<void>::Ok();
+      });
+}  // namespace
+}  // namespace oneflow
+```
+
+Analyze the above code: 
+
+- The `oneflow/core/framework/framework.h` contains all the interfaces we need to create an Op
+- `.Attr<std::string>("device_sub_tag", "py")` is necessary which tells OneFlow to call the Python Kernel when using this Op.
+- The interface related to the custom op is organized in `oneflow::user_op`. Using the namespace `oneflow` to simplify the type name.
+- The macro `REGISTER_USER_OP` is used to register Op which accepts an parameter `user_relu_forward` as `op_type_name`.
+- After registering with `REGISTER_USER_OP`, it actually returns an `OpRegistry` class (located in [user_op_registry.h]()). By calling this class, the setting of a custom op is accomplished:
+    1. `Input("in") ` indicates it has an input named "in".
+    2. `Output("out") ` indicates that it has an output named "out".
+    3. `SetTensorDescInferFn` is used to set the shape and data type derivation function. Also to describe the relationship between the shape and type of the output of the operator and the input. In the above code, the output shape and data type are same as the input.
+
+The `op_type_name_cpp_def.cpp` is the only C++ file that will be used in the implementation of the `Python Kernel`. It is used to configure information of Op. At this stage, it is not possible to skip the step of configuring Op using C++ (since it is necessary for setting advanced information such as distributions). But as we can see, the file does not involve specific operations. Even if you are not familiar with C++, you can easily master it according to our examples.
```

Comment:
```suggestion
The `op_type_name_cpp_def.cpp` is the only C++ file that will be used in the implementation of the `Python Kernel`. It is used to configure information of Op. By far, we cannot remove the C++ configuration of Op since it is necessary for setting advanced information such as distributions. But as we can see, that file does not involve specific operations, so, even you are not familiar with C++, you can easily master it according to our examples.
```